### ISSUE TITLE
bug: Ionic Slide Box Breaks on splice an item. Goes to last index and overlaps. 

### ISSUE THREAD

**Issue by scripterkaran** [driftyco/ionic#3431]
**Type**: <span ionic-type>bug</span>

**Platform**: <span ionic-platform>ios</span> <span ionic-platform-version>8</span>  <span ionic-webview>webview</span>

<span ionic-description>Here is the code pen that shows the error. Try removing the item , example product 3, the slide box slides to last product and overlaps the other items. ( see the pager, it moves ) 

http://codepen.io/anon/pen/qEvyLV?editors=101

any help would be big help. </span>

EDIT: I included a $ionicSlideBoxDelegate.update(); on slide change function, it seems to slightly fix the issue but only on slide change. Its not a 100% solution. Check Code pen.
<span is-issue-template></span>

**Comment by Ionitron** [comment]
Greetings @promactkaran!

My sensors indicate that you need to update your issue through our [custom issue form](http://ionicframework.com/submit-issue/#/?iid=3431). We are now requiring all issues to be submitted this way, to ensure that we have all of the information necessary to fix them as quickly as possible.

**[Click Here To Update Your Issue](http://ionicframework.com/submit-issue/#/?iid=3431)**

I will have no choice but to close this issue if it is not resubmitted through the form. Please fill out the rest of the form, so that I may use my friendly robot powers to assist you.

Thank you!

<span ionitron-issue-resubmit></span>

**Comment by agnellvj** [comment]
The slidebox implementation is broken when you try to update the underlying collection.

Just another demonstration of it. Also, my last comment links to a branch that has not been included in the RC's yet that contains an overhaul of the slidebox implementation.
https://github.com/driftyco/ionic/issues/3076

**Comment by perrygovier** [comment]
If you give the slidebox a millisecond or two for some asynchronous actions, it looks to work properly:
http://codepen.io/perrygovier/pen/QwoPbb?editors=101 

Let me know if that doesn't resolve the issue

**Comment by promactkaran** [comment]
This is a good try, but unfortunately, its all messy now also, if I splice 1st product it goes to product3, and one major problem , if I splice last object (product 11) it splices behind leaving the screen blank..  @perrygovier

**Comment by promactkaran** [comment]
Possible work around I found to this problem: setting a SetInterval javascript function. 
though its not 100% correct , it is one of the workaround when you want to splice slidebox slides. 

here is a code pen for the same: 
http://codepen.io/anon/pen/qEvyLV?editors=101

**Comment by CodyMorris** [comment]
I'm having a very similar issue on Android as well, Looks like some users (including me) found this closed issue and are posting there:

https://github.com/driftyco/ionic/issues/1578#issuecomment-74268456

**Comment by agnellvj** [comment]
@CodyMorris have you tried the suggestion of calling update multiple times in a $timeout? It is the only known workaround right now with what is released. I use a loop that runs every 250ms for 4 times just to be sure.

**Comment by promactkaran** [comment]
@agnellvj Yes thats exactly what I do right now. I have set out a Interval function of javascript to call update every 200ms for 2 seconds.

**Comment by omprakash1989** [comment]
Is there any fix of this issue? Please update.

**Comment by CodyMorris** [comment]
@agnellvj yeah, that works but its definitely not ideal

**Comment by omprakash1989** [comment]
Well...This worked for me though this is not a solution of the bug.
$scope.trashMe = function(index) {
  var indexToremove = $ionicSlideBoxDelegate.currentIndex();
  $scope.next();
  $timeout(function() {
    var data = $scope.slideData;
    data.splice(indexToremove, 1);
    $scope.slideData = data;
    $ionicSlideBoxDelegate.update();
  });
  $scope.previous();
};

**Comment by labrute974** [comment]
Hey guys, look at PR: https://github.com/driftyco/ionic/pull/3842

It's all broken because of angular ngAnimate, disabling it on the ionSlideBox directive works.

**Comment by craigvl** [comment]
Having same issue.

**Comment by yesir1006** [comment]
Any progress in this issue, still facing the problem.

**Comment by xudao520** [comment]
having same problem, do you solved the problem?

**Comment by mlynch** [comment]
Hey everyone, can you try the new slidebox just pushed to master (available in upcoming 1.2):

```html
<ion-slides options="options">
  <ion-slide-page>
  </ion-slide-page>
</ion-slides>
```

**Comment by mlynch** [comment]
New slidebox should fix this ,closing until I get confirmation
